docs/rules/no-reject-literal.md

```diff
@@ -0,0 +1,57 @@
+# disallow using literals as Promise rejection reasons (no-reject-literal)
+
+It is considered good practice to only instances of the built-in `Error` object for user-defined errors in Promises. `Error` objects automatically store a stack trace, which can be used to debug an error by determining where it came from. If a Promise is rejected with a non-`Error` value, it can be difficult to determine where the rejection occurred.
+
+
+## Rule Details
+
+This rule aims to ensure that Promises are only rejected with `Error` objects.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint no-reject-literal: "error"*/
+
+Promise.reject("something bad happened");
+
+Promise.reject(5);
+
+Promise.reject();
```

Comment:
This should be totally fine - rejecting with a literal is a problem, but rejecting with nothing (simply to be able to trigger a `.catch`) is totally legit imo.

At the very least, could this be an option, so that I can use this rule while allowing `Promise.reject()`?